### PR TITLE
Enable running tests on full .NET Framework version of MSBuild

### PR DESCRIPTION
**Pull request by dsplaisted**
- Add `-FullMSBuild` build switch to run tests using full MSBuild
- Add CI configuratons to include full MSBuild runs

### CODE REVIEW

**Comment by dsplaisted** [comment]
@dotnet-bot test CI please

**Comment by dsplaisted** [comment]
@dotnet-bot test CI please

**Comment by dsplaisted** [comment]
@dotnet-bot retest Windows_NT Debug please

**Comment by dsplaisted** [comment]
@nguerrera @333fred @srivatsn @natidea for review

@tannergooding @dotnet/roslyn-infrastructure for review since this is an infrastructure change

**Review by natidea — APPROVED** [review]

**Review by 333fred — APPROVED** [review]
LGTM, but I'd still like someone from @dotnet/roslyn-infrastructure to review.

**Review by srivatsn — APPROVED** [review]

File: test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibrary.cs [hunk 1]
```diff
@@ -76,6 +76,12 @@ public void All_props_and_targets_add_themselves_to_MSBuildAllTargets()
                 return;
             }
 
+            //  Disable this test when using full Framework MSBuild, as the paths to the props and targets are different
+            if (UsingFullFrameworkMSBuild)
```
Comment (tannergooding) [inline test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibrary.cs]: I would much rather we create a custom attribute and skip this dynamically: https://github.com/xunit/samples.xunit/blob/master/DynamicSkipExample

File: test/Microsoft.NET.Build.Tests/GivenThereAreDefaultItems.cs [hunk 2]
```diff
@@ -370,6 +370,13 @@ public void Default_items_have_the_correct_relative_paths()
         [Fact]
         public void Compile_items_can_be_explicitly_specified_while_default_EmbeddedResource_items_are_used()
         {
+            if (UsingFullFrameworkMSBuild)
```
Comment (tannergooding) [inline test/Microsoft.NET.Build.Tests/GivenThereAreDefaultItems.cs]: Same here, would much rather this be dynamically skipped than falsely marking it as passing

**Review by tannergooding — APPROVED** [review]
Overall this LGTM. A couple minor nits and a concern about marking some tests as "passing" when we should be dynamically skipping them and aiming to support them eventually (and even running them if we do have admin privileges for the current run).

File: test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibrary.cs (hunk 1 above)
Comment (dsplaisted) [inline test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibrary.cs]: We're currently doing it like this to skip tests on non-Windows, and perhaps in other cases.  I've filed #736 for actually marking these as skipped.

**Pull request merged**